lib/util/http/index.ts

```diff
@@ -92,6 +92,33 @@ export class Http {
     return { ...res, body };
   }
 
+  async patchJson<T = unknown>(
+    url: string,
+    options: HttpPostOptions
+  ): Promise<HttpResponse<T>> {
+    const res = await this.request(url, { ...options, method: 'patch' });
+    const body = is.string(res.body) ? JSON.parse(res.body) : res.body;
+    return { ...res, body };
+  }
+
+  async putJson<T = unknown>(
+    url: string,
+    options: HttpPostOptions
+  ): Promise<HttpResponse<T>> {
+    const res = await this.request(url, { ...options, method: 'put' });
+    const body = is.string(res.body) ? JSON.parse(res.body) : res.body;
+    return { ...res, body };
+  }
+
+  async deleteJson<T = unknown>(
+    url: string,
+    options: HttpPostOptions
+  ): Promise<HttpResponse<T>> {
+    const res = await this.request(url, { ...options, method: 'delete' });
+    const body = is.string(res.body) ? JSON.parse(res.body) : res.body;
```

Comment:
I guess we can de-dupe this by adding a private function. like `this.requestJson()`?